packages/babel-parser/src/parser/statement.js

```diff
@@ -2113,45 +2114,24 @@ export default class StatementParser extends ExpressionParser {
     node.specifiers.push(this.finishNode(specifier, type));
   }
 
-  maybeParseModuleAttributes() {
-    if (this.match(tt._with) && !this.hasPrecedingLineBreak()) {
-      this.expectPlugin("moduleAttributes");
-      this.next();
-    } else {
-      if (this.hasPlugin("moduleAttributes")) return [];
-      return null;
-    }
+  parseAssertEntries() {
+    this.expectPlugin("importAssertions");
+
     const attrs = [];
-    const attributes = new Set();
+
     do {
-      // we are trying to parse a node which has the following syntax
-      // with type: "json"
-      // [with -> keyword], [type -> Identifier], [":" -> token for colon], ["json" -> StringLiteral]
       const node = this.startNode();
-      node.key = this.parseIdentifier(true);
-
-      // for now we are only allowing `type` as the only allowed module attribute
-      if (node.key.name !== "type") {
-        this.raise(
-          node.key.start,
-          Errors.ModuleAttributeDifferentFromType,
-          node.key.name,
-        );
-      }
 
-      // check if we already have an entry for an attribute
-      // if a duplicate entry is found, throw an error
-      // for now this logic will come into play only when someone declares `type` twice
-      if (attributes.has(node.key.name)) {
-        this.raise(
-          node.key.start,
-          Errors.ModuleAttributesWithDuplicateKeys,
-          node.key.name,
-        );
+      // parse AssertionKey : IdentifierName, StringLiteral
+      let assertionKeyNode;
+      if (this.match(tt.string)) {
+        assertionKeyNode = this.parseLiteral(this.state.value, "StringLiteral");
+      } else {
+        assertionKeyNode = this.parseIdentifier();
       }
-      attributes.add(node.key.name);
-      this.expect(tt.colon);
-      // check if the value set to the module attribute is a string as we only allow string literals
+      this.next();
+      node.key = assertionKeyNode;
+
```

Comment:
just noticed that i'm missing the ModuleAttributesWithDuplicateKeys check now. will add again